src/modules/twap/utils/updateTwapOrdersList.ts

```diff
@@ -0,0 +1,17 @@
+import { TWAP_NOT_PENDING_STATUSES } from '../const'
+import { TwapOrdersList } from '../state/twapOrdersListAtom'
+
+export function updateTwapOrdersList(currentState: TwapOrdersList, nextState: TwapOrdersList): TwapOrdersList {
+  const newState = { ...currentState, ...nextState }
+
+  Object.keys(nextState).forEach((orderId) => {
+    const currentOrder = currentState[orderId]
+
+    // Don't update orders in final statuses
+    if (currentOrder && TWAP_NOT_PENDING_STATUSES.includes(currentOrder.status)) {
+      newState[orderId] = currentOrder
+    }
+  })
```

Comment:
I don't get what's happening here

First, the parameter name is `...List` but it's actually an object

Then, both parameters are combined into a single object

Later on line `12` the given order in the new obj is updated with data from `currentState`?

Why do you need to merge both?
Wouldn't it be enough to just iterate over `currentState` keys?

Reply:
Agree, this code is odd.
Refactored it and added comments:
https://github.com/cowprotocol/cowswap/pull/2633/commits/c71508ac09e854eebbd8b1653aeffa54ec8816d7